lib/helpers/calculator_helpers.js

```diff
@@ -0,0 +1,471 @@
+        observationDefs.forEach((obDef) => {
+          if (!populationResults.values) {
+            populationResults.values = [];
+          }
+          // Observations only have one result, based on how the HQMF is
+          // structured (note the single 'value' section in the
+          // measureObservationDefinition clause).
+          const obsResults =
+            results.patientResults != null && results.patientResults[patientId] != null
+              ? results.patientResults[patientId][obDef]
+              : undefined;
+
+          obsResults.forEach((obsResult) => {
+            // Add the single result value to the values array on the results of
+            // this calculation (allowing for more than one possible observation).
+            if (obsResult != null ? Object.prototype.hasOwnProperty.call(obsResult, 'value') : undefined) {
+              // If result is a Cql.Quantity type, add its value
+              populationResults.values.push(obsResult.observation.value);
+            } else {
+              // In all other cases, add result
+              populationResults.values.push(obsResult.observation);
+            }
+          });
+        });
+      }
+    });
+    return populationResults;
+  }
+
+  /**
+   * Create population values (aka results) for all episodes using the results from the calculator. This is
+   * used only for the episode of care measures
+   * @param {Measure} measure - The measure of the population set
+   * @param {Population} population - The population set we are getting the values for.
+   * @param {object} results - The raw results object from the calculation engine.
+   * @param {Patient} patientId - The patient ID we are getting results for.
+   * @param {Array} observationDefs - List of observation defines we add to the elm for calculation OBSERVs.
+   * @returns {object} The episode results. Map of episode id to population results which is a map of "POPNAME"
+   * to Integer result. Except for OBSERVs, their key is 'value' and value is an array of results.
+   */
+  static createEpisodePopulationValues(measure, population, results, patientId, observationDefs, populationIndex) {
+    const episodeResults = {};
+    // Grab the mapping between populations and CQL statements
+    const cqlMap = measure.populations_cql_map;
+    // Loop over all population codes ("IPP", "DENOM", etc.) to deterine ones included in this population.
+    const popCodesInPopulation = [];
+    Measure.ALL_POPULATION_CODES.forEach((popCode) => {
+      if (population[popCode] != null) {
+        popCodesInPopulation.push(popCode);
+      }
+    });
+
+    popCodesInPopulation.forEach((popCode) => {
+      let newEpisode;
+      let values;
+      if (cqlMap[popCode]) {
+        const definedPops = cqlMap[popCode];
+        let cqlPopulation = definedPops[populationIndex];
+        if (population.population_index != null && popCode !== 'STRAT') {
```

Comment:
Can this block be pulled out into a separate function in this file, as it is repeated in createPatientPopulationValues?